modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -109,39 +115,50 @@ class GmosControllerEpics[T<:GmosController.SiteDependentTypes](encoders: GmosCo
     _ <- DC.setCcdYBinning(encode(dc.bi.y))
   } yield ()
 
-
+  private def smartSetParam[A: Equal](v: A, get: => Option[A], set: SeqAction[Unit]): SeqAction[Unit] =
+    get.flatMap(g => (g =/= v).option(set)).getOrElse(SeqAction.void)
```

Comment:
I first defined `set` as `A => SeqAction[Unit]`, and passed `v` to it. The problem is that, in some cases, the type of the parameter is not always the type of the associated status.

Reply:
Looks nice